cumulusci/core/keychain/base_project_keychain.py

```diff
@@ -223,7 +236,12 @@ def get_service(self, service_type, alias=None):
         return service
 
     def _get_service(self, service_type, alias):
-        return self.services[service_type][alias]
+        try:
+            return self.services[service_type][alias]
+        except KeyError:
+            raise ServiceNotConfigured(
+                f"No service of type {service_type} configured with name: {alias}"
+            )
```

Comment:
This was needed for tests in `test_connectedapp.py` as that test class instantiates the `BaseProjectKeychain` instead of `EncryptedFileProjectKeychain`. This shouldn't affect anything as `BaseProjectKeychain` is not used on its own and `EncryptedFileProjectKeychain` implements its own `_get_service()` method.